src/api/repos/user-repo.ts

```diff
@@ -140,4 +141,126 @@ const submitQuiz = async (id: string, quiz: Quiz) => {
+    return client.createToken(userId);
+};
+
+const createChat = async (userId: string, channelId: string, channelName: string) => {
+    // @ts-ignore
+    const client = StreamChat.getInstance(process.env.STREAM_CHAT_API_KEY, process.env.STREAM_CHAT_API_SECRET);
+
+    const channel = client.channel('messaging', channelId, {
+        created_by_id: userId,
+        name: channelName
+    });
+    await channel.create()
+      .catch((err) => createNewObjectCaughtError(err, 'createChat', 'There has been an error in creating a new chat'));
+
+    // Add the user that created this channel as a member otherwise the channel will not appear in his list on the mobile app
+    await channel.addMembers([userId])
+      .catch((err) => createNewObjectCaughtError(err, 'createChat', 'There has been an error in adding the user as a member into the newly created chat'));
+
+    return httpStatusCode.CREATED;
+};
+
+const deleteActivityChat = async (channelId: string) => {
+    // @ts-ignore
+    const client = StreamChat.getInstance(process.env.STREAM_CHAT_API_KEY, process.env.STREAM_CHAT_API_SECRET);
+    const channel = client.channel('messaging', channelId);
+    await channel.delete().catch((err) => createNewObjectCaughtError(err, 'deleteActivityChat', 'There has been an error in deleting a chat'));
+    return httpStatusCode.DELETED;
+};
+
+const addMemberToActivityChat = async (user_id: string, user_name: string, channel_id: string) => {
+    // @ts-ignore
+    const client = StreamChat.getInstance(process.env.STREAM_CHAT_API_KEY, process.env.STREAM_CHAT_API_SECRET);
+    const channel = client.channel('messaging', channel_id);
+    await channel.addMembers([user_id], { text: user_name + ' joined the channel.', user_id: user_id })
+      .catch((err) => createNewObjectCaughtError(err, 'addMemberToActivityChat', 'There has been an error in adding a member to a chat'));
+    return httpStatusCode.OK;
+};
+
+const removeMemberFromActivityChat = async (user_id: string,channel_id: string) => {
+    // @ts-ignore
+    const client = StreamChat.getInstance(process.env.STREAM_CHAT_API_KEY, process.env.STREAM_CHAT_API_SECRET);
+    const channel = client.channel('messaging', channel_id);
+    await channel.removeMembers([user_id])
+      .catch((err) => createNewObjectCaughtError(err, 'removeMemberFromActivityChat', 'There has been an error in removing a member from the chat'));
+    return httpStatusCode.OK;
+};
+
+const createNewStreamChatUser = async (user_id: string, user_name: string) => {
+    // @ts-ignore
+    const client = StreamChat.getInstance(process.env.STREAM_CHAT_API_KEY, process.env.STREAM_CHAT_API_SECRET);
+    await client.upsertUser({
+        id: user_id,
+        name: user_name
+     }).catch((err) => createNewObjectCaughtError(err, 'createNewStreamChatUser', 'There has been an error in creating a new Stream Chat user'));
+    return httpStatusCode.CREATED;
+};
+
```

Comment:
lots of `// ts-ignore` without explanation there. you probably know what they're for but would be nice if there's something.